### ISSUE TITLE
<filesystem>: Cannot remove files on one specific machine

### ISSUE THREAD

**Issue by johannes-inmation** [microsoft/STL#2224]
**Description**
The `filesystem::remove` function is not able to remove files and fails with the error message "Incorrect function.".
This only happens on one specific VM which is on the following windows version:
```
Edition: Windows Server 2016 Standard
Version: 1607
OS Build: 14393.1884
```
The issue does not occur on a different Windows Server 2016 VM.
The issue also does not occur on the problematic VM if the program is compiled with the x86 compiler.
In the code of the `filesystem::remove` function, I have seen that there is a special case for Windows 10 1607, which handles the `__std_win_error::_Invalid_function` error: https://github.com/microsoft/STL/blob/78ff4618d4fa43eeb4bb714d76f27d88cf11a1b6/stl/src/filesystem.cpp#L156

I am able to remove the file without any issues by using the `_wremove` function instead of `filesystem::remove`.

Are you able to provide further information what is causing this issue?

**Command-line test case**
```
C:\Temp>type repro.cpp
#include <iostream>
#include <filesystem>
int main(int argc, char *argv[], char *envp[])
{
	std::error_code errc;
	std::filesystem::remove(argv[1], errc);
	if (errc)
		std::cout << argv[1] << " " << errc.message() << std::endl;
	else
		std::cout << "Success" << std::endl;
}

C:\Temp>cl /EHsc /std:c++latest repro.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.16.27045 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

repro.cpp
Microsoft (R) Incremental Linker Version 14.16.27045.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:repro.exe
repro.obj

```
On Windows Server 2016 Standard VM
```
C:\Temp>echo test > test.txt
C:\Temp>.\repro.exe .\test.txt
.\test.txt Incorrect function.
```

**Expected behavior**
The file test.txt should be delete and the program should output "Success".

**STL version**
```
Microsoft Visual Studio Community 2017
Version 15.9.39
```

**Comment by billy-baker** [comment]
Even though the DeveloperCommunity issue from 2018 (link below) was closed as not a bug, remove was fixed in 2019 (16.0). 

https://developercommunity.visualstudio.com/t/std::filesysetm::remove-fails-on-WIn10-1/338783?space=62&q=std%3A%3Afilesysetm%3A%3Aremove+fails+on+WIn10+14393

**Comment by strega-nil-ms** [comment]
Since this bug does not meet the servicing bar, and it has been fixed in VS2019, I'm closing this as resolved. Thanks for the report @johannes-inmation!
